File: pyiArduinoI2Cencoder/cpp/PositionSettings.cpp
// ПРИМЕР НАСТРОЙКИ ПОКАЗАНИЙ ПОЗИЦИИ ВАЛА:

#include <cstdio>

// Подключаем библиотеку для работы с энкодером I2C-flash.
#include "../iarduino_I2C_Encoder.h"

// Инстанциируем объект, указывая адрес модуля на шине I2C.
iarduino_I2C_Encoder enc(0x09);

void loop(void);
int main()
{
	// Инициируем работу с энкодером.
	enc.begin();
	// Указываем считать до 5 полных оборотов, без отрицательных значений.
	enc.setPosSettings(5, false);
	for (;;)
		loop();
}

void loop()
{
	int16_t p = enc.getPosition();
	// Выводим текущую позицию вала энкодера.
	printf("%d\n", p);
	delay(500);
}

File: pyiArduinoI2Cencoder/examples/PositionSettings.py
# ПРИМЕР НАСТРОЙКИ ПОКАЗАНИЙ ПОЗИЦИИ ВАЛА:

from time import sleep

# Подключаем библиотеку для работы с энкодером I2C-flash.
from pyiArduinoI2Cencoder import *

# Инстанцируем объект, указывая адрес модуля на шине I2C.
enc = pyiArduinoI2Cencoder(0x09)

# Указываем считать до 5 полных оборотов, без отрицательных значений.
enc.setPosSettings(5, False)

while True:

    p = enc.getPosition()
    # Выводим текущую позицию вала энкодера.
    print(p)
    sleep(.5)

